compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUTileToVectorSize.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h"
+#include "iree/compiler/Codegen/LLVMCPU/Passes.h"
+#include "iree/compiler/Codegen/LLVMCPU/Utils.h"
+#include "llvm/Support/DebugLog.h"
+#include "llvm/Support/InterleavedRange.h"
+#include "llvm/Support/LogicalResult.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/MemRef/Transforms/Transforms.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Patterns.h"
+#include "mlir/Dialect/SCF/Transforms/TileUsingInterface.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/BuiltinTypeInterfaces.h"
+#include "mlir/IR/Iterators.h"
+#include "mlir/Interfaces/TilingInterface.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-llvmcpu-tile-to-vector-size"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_LLVMCPUTILETOVECTORSIZEPASS
+#include "iree/compiler/Codegen/LLVMCPU/Passes.h.inc"
+
+namespace {
+
+struct LLVMCPUTileToVectorSizePass final
+    : impl::LLVMCPUTileToVectorSizePassBase<LLVMCPUTileToVectorSizePass> {
+  using impl::LLVMCPUTileToVectorSizePassBase<
+      LLVMCPUTileToVectorSizePass>::LLVMCPUTileToVectorSizePassBase;
+
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<arith::ArithDialect, scf::SCFDialect>();
+  }
+
+  void runOnOperation() override;
+};
+
+static std::optional<SmallVector<int64_t>>
+getTileSizesForEachDims(linalg::LinalgOp op) {
+  IREE::Codegen::LoweringConfigAttrInterface loweringConfig =
+      getLoweringConfig(op);
+  SmallVector<bool> scalableFlags = loweringConfig.getVectorScalableFlags();
+  if (llvm::count(scalableFlags, true) > 0) {
+    return std::nullopt;
+  }
+
+  unsigned int numLoops = op.getNumLoops();
```

Comment:
```suggestion
  unsigned numLoops = op.getNumLoops();
```
Isn't this the same and shorter?

Reply:
Yes, thanks.